jujugui/static/gui/src/app/views/ghost-deployer-extension.js

```diff
@@ -74,12 +74,14 @@ YUI.add('ghost-deployer-extension', function(Y) {
       var serviceName = ghostService.get('name');
       var charmId = this._addSeriesToCharmId(charm.get('id'), activeSeries);
       var constraints = {};
-      // TODO frankban: add support for fetching delegatable macaroons that can
-      // be used to add private charms.
-      this.env.addCharm(
-        charmId, null, this._addCharmCallbackHandler.bind(this, charm),
-        // Options used by ECS, ignored by environment.
-        {applicationId: ghostServiceId});
+      if (this.isLegacyJuju() || charm.get('id').indexOf('local:') === -1) {
```

Comment:
I think the idea here is that if a charm is local then it's been already added (via the HTTP API) and so we don't need to add it again with AddCharm (which uses the charm store and does not support local charms). AFAIK this is not based on the Juju version being used. So I am not sure about the isLegacyJuju() check, I'd just skip adding the charm `if charm.get('id').startsWitth('loca')`.